service/main.ml

```diff
@@ -26,56 +26,68 @@ let has_role user = function
-            report = {|\1|};
-            score = 40;
-          };
-          { (* Generic error caught by opam (e.g. uncaught OCaml exception) *)
-            pattern = {|[\n]# Fatal error: (.+)[\n]|};
-            report = {|\1|};
-            score = 35;
-          };
-          { (* Generic errors caught by opam (e.g. gcc) *)
-            pattern = {|[\n]# .+: error: (.+)[\n]|};
-            report = {|\1|};
-            score = 30;
-          };
-        ]
-      in
-      List.iter Current.Log_matcher.add_rule default_rules
+  let default_rules =
+    let open Current.Log_matcher in
+    [
+      { (* Opam when the package or one of its dependencies has available: <non-compatible-condition> *)
+        pattern = {|[\n]\[ERROR\] .+ unmet availability conditions: .+[\n]|};
+        report = {|[SKIP] Package not available|};
+        score = 100;
+      };
+      { (* Opam 2.0 when the package or one of its dependencies isn't available on the current switch/plateform *)
+        pattern = {|[\n]\[ERROR\] No solution for .+: The following dependencies couldn't be met:[\n]|};
+        report = {|[SKIP] Package not available|};
+        score = 100;
+      };
+      { (* Opam 2.1 when the package or one of its dependencies isn't available on the current switch/plateform *)
+        pattern = {|[\n]\[ERROR\] Package conflict![\n]|};
+        report = {|[SKIP] Package not available|};
+        score = 100;
+      };
+      { (* Ignore failures on failing packages when the platform being tested is contained in the x-ci-accept-failures optional field *)
+        (* See lib/opam_build.ml corresponding bash command printing this exact line *)
+        pattern = {|[\n]A package failed and has been disabled for CI using the 'x-ci-accept-failures' field\.[\n]|};
+        report = {|[SKIP] Failure ignored|};
+        score = 100;
+      };
+      { (* e.g. build: ["bash"] on platforms without bash will result in this error *)
+        pattern = {|[\n]# bwrap: execvp (.+): No such file or directory[\n]|};
+        report = {|\1 not found|};
+        score = 50;
+      };
+      { (* OCaml errors *)
+        pattern = {|[\n]# Error: (.+)[\n]|};
+        report = {|\1|};
+        score = 48;
+      };
+      { (* OCaml Exceptions *)
+        pattern = {|[\n]# Exception: (.+)[\n]|};
+        report = {|\1|};
+        score = 45;
+      };
+      { (* Generic errors caught by opam (e.g. cargo) *)
+        pattern = {|[\n]# error: (.+)[\n]|};
+        report = {|\1|};
+        score = 40;
+      };
+      { (* Generic error caught by opam (e.g. uncaught OCaml exception) *)
+        pattern = {|[\n]# Fatal error: (.+)[\n]|};
+        report = {|\1|};
+        score = 35;
+      };
+      { (* Generic errors caught by opam (e.g. gcc) *)
+        pattern = {|[\n]# .+: error: (.+)[\n]|};
+        report = {|\1|};
+        score = 30;
+      };
+      { (* Opam errors *)
+        pattern = {|[\n]\[ERROR\] (.+)[\n]|};
```

Comment:
for some reason this pattern seems to eliminate all other ones with `[ERROR]` even with a higher score..

Reply:
Perhaps the others don't have the `\n` at the start? Once there is a match starting at the `\n` I think it will skip ahead to after the matched text looking for the next one.

Reply:
ah :/ Then I suppose there is a bug when there are several regexp matching containing `\n`.

All the patterns have `\n` at the beginning:
<img width="923" alt="Screenshot 2022-04-19 at 11 58 30" src="https://user-images.githubusercontent.com/2611789/163989482-5b8ec632-bda4-47e0-be1d-436232ffe060.png">